resource/js/components/PageEditor/ContextBasedNewLineHandlerExecutor.js

```diff
@@ -0,0 +1,29 @@
+import * as codemirror from 'codemirror';
+
+import markdownListHelper from './MarkdownListHelper';
+import markdownTableHelper from './MarkdownTableHelper';
+
+/**
+ * Selector for one new line handler
+ */
+class ContextBasedNewLineHandlerExecutor {
+
+  /**
+   * select one handler from helper
+   * @param {any} editor An editor instance of CodeMirror
+   */
+  execNewLineHandler(editor) {
+    let newLineHelpers = [markdownTableHelper, markdownListHelper];
+    const helper = newLineHelpers.find( h => h.isMatchedContext(editor));
+    if (helper != undefined) {
+      helper.handleNewLine(editor);
```

Comment:
This process only the first matched helper.
How to handle multiple helpers that either matche in future?

Interceptor mechanism can handle multiple interceptors when an arbitrary event fires.

Reply:
Uh... I thought better that it forced to handle by only one helper than by multiple helpers.

Because if it allowed multiple helpers to handle, when a helper replaces markdown text, it causes precondition of the other helper will change and complex so on.

I try to refer interceptor mechanism and rethink.

Thanks.